runtime/domain.c

```diff
@@ -417,50 +414,52 @@ static uintnat fresh_domain_unique_id(void) {
 /* must be run on the domain's thread */
 static void create_domain(uintnat initial_minor_heap_wsize) {
   dom_internal* d = 0;
+  caml_domain_state* domain_state;
+  struct interruptor* s;
+
   CAMLassert (domain_self == 0);
 
   /* take the all_domains_lock so that we can alter the STW participant
      set atomically */
   caml_plat_lock(&all_domains_lock);
 
   /* wait until any in-progress STW sections end */
-  while (atomic_load_acq(&stw_leader)) caml_plat_wait(&all_domains_cond);
+  while (atomic_load_acq(&stw_leader))
+    caml_plat_wait(&all_domains_cond);
 
   d = next_free_domain();
   if (d) {
-    struct interruptor* s = &d->interruptor;
+    s = &d->interruptor;
     CAMLassert(!s->running);
     CAMLassert(!s->interrupt_pending);
-    if (!s->interrupt_word) {
-      caml_domain_state* domain_state;
-      atomic_uintnat* young_limit;
-      /* never been started before, so set up minor heap */
-      if (!caml_mem_commit(
-              (void*)d->tls_area, (d->tls_area_end - d->tls_area))) {
-        /* give up now */
-        d = 0;
+
+    domain_self = d;
+
+    /* If the chosen domain slot has not been previously used, allocate a fresh
+     * domain state. Otherwise, reuse it. Reusing the slot ensures that the GC
+     * stats are not lost. */
+    if (!d->state) {
+      /* FIXME: Never freed. Not clear when to. */
+      domain_state = (caml_domain_state*)
+        caml_stat_calloc_noexc(1, sizeof(caml_domain_state));
```

Comment:
If I understand correctly, this is not actually a leak, at least in "being nice with shutdown mode" because `caml_stat_calloc_noexc` is pooled and will be properly reclaimed by `caml_shutdown`.

Why don't we have a `caml_free_domains` function that would be de-allocating parent of `caml_init_domains`, and would be called by `caml_shutdown`? Is this simply an omission, something that would be useless, or an actively bad idea for some reason I am missing?